### PR TITLE
allow derived types of DataFrame

### PR DESCRIPTION
**Pull request by liangfu**
This PR updates DataFrame type checkings to allow derived types of DataFrame as input type.

### CODE REVIEW

**Comment by liangfu** [comment]
@microsoft-github-policy-service agree [company="Amazon"]

**Comment by liangfu** [comment]
```
@microsoft-github-policy-service agree company="Amazon"
```

**Comment by liangfu** [comment]
@microsoft-github-policy-service agree

**Comment by liangfu** [comment]
@microsoft-github-policy-service agree company="Amazon"

File: hummingbird/ml/_executor.py [hunk 1]
```diff
@@ -67,15 +67,15 @@ def _fix_var_naming(operators, names, mod="input"):
...
+                isinstance(inputs[0], DataFrame)
                 and DataFrame is not None
                 and not self.check_dataframe_to_array
                 and len(self._input_names) == len(inputs[0].columns)
             ), "number of inputs or number of columns in the dataframe do not match with the expected number of inputs {}".format(
                 self._input_names
             )
 
-            if type(inputs[0]) == DataFrame and DataFrame is not None:
+            if isinstance(inputs[0], DataFrame) and DataFrame is not None:
```
Comment (interesaaat) [inline hummingbird/ml/_executor.py]: Maybe `if  DataFrame is not None and isinstance(inputs[0], DataFrame)` will solve the test errors.

Comment (liangfu) [reply hummingbird/ml/_executor.py]: Fixed.

Comment (interesaaat) [reply hummingbird/ml/_executor.py]: I still see some test error somewhere. Let me check what is going on.

**Review by interesaaat — APPROVED** [review]
LGTM

**Pull request merged**